build/OutputDirectories.props

```diff
@@ -1,20 +1,29 @@
 <Project>
   <PropertyGroup>
-      <BaseOutputDirectory>$(RepoRoot)/artifacts/$(Rid)</BaseOutputDirectory>
-      <OutputDirectory>$(BaseOutputDirectory)/stage2</OutputDirectory>
-      <Stage2WithBackwardsCompatibleRuntimesOutputDirectory>$(BaseOutputDirectory)/stage2WithBackwardsCompatibleRuntimes</Stage2WithBackwardsCompatibleRuntimesOutputDirectory>
+      <CliBuildStage Condition="'$(CliBuildStage)' == ''">2</CliBuildStage>
+
+      <!-- BaseOutputDirectory is the root output path for this stage and RID
+           BaseOutputPath is the project-specific output folder that each project builds into.
+           This is not confusing at all. -->
+      <BaseOutputDirectory>$(RepoRoot)/artifacts/stage$(CliBuildStage)/$(Rid)</BaseOutputDirectory>
+      <BaseOutputPath>$([System.IO.Path]::GetFullPath('$(BaseOutputDirectory)/bin/$(MSBuildProjectName)'))</BaseOutputPath>
+
+      <BaseIntermediateOutputPath>$([System.IO.Path]::GetFullPath('$(BaseOutputDirectory)/obj/$(MSBuildProjectName)'))</BaseIntermediateOutputPath>
+      <OutputDirectory>$(BaseOutputDirectory)/dotnet</OutputDirectory>
+      <Stage2WithBackwardsCompatibleRuntimesOutputDirectory>$(BaseOutputDirectory)/dotnetWithBackwardsCompatibleRuntimes</Stage2WithBackwardsCompatibleRuntimesOutputDirectory>
```

Comment:
I see these are also set in signing.proj. Will the mismatched values break signing? Can you get rid of the duplication?

Reply:
https://github.com/dotnet/cli/blob/8ef162fd13005a56edf8729cdd91a76eaac0309d/build/Signing.proj#L16-L20